packages/lit-html/src/lib/directives/live.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {
+  directive,
+  NodePart,
+  Directive,
+  Part,
+  EventPart,
+  BooleanAttributePart,
+  AttributePart,
+  PropertyPart,
+  noChange,
+  nothing,
+} from '../lit-html.js';
+
+/**
+ * Checks binding values against live DOM values, instead of previously bound
+ * values, when determining whether to update the value.
+ *
+ * This is useful for cases where the DOM value may change from outside of
+ * lit-html, such as with a binding to an `<input>` element's `value` property,
+ * a content editable elements text, or to a custom element that changes it's
+ * own properties or attributes.
+ *
+ * In these cases if the DOM value changes, but the value set through lit-html
+ * bindings hasn't, lit-html won't know to update the DOM value and will leave
+ * it alone. If this is not what you want—if you want to overwrite the DOM
+ * value with the bound value no matter what—use the `live()` directive:
+ *
+ *     html`<input .value=${live(x)}>`
+ *
+ * `live()` performs a strict equality check agains the live DOM value, and if
+ * the new value is equal to the live value, does nothing. This means that
+ * `live()` should not be used when the binding will cause a type conversion. If
+ * you use `live()` with an attribute binding, make sure that only strings are
+ * passed in, or the binding will update every render.
+ */
+
+class LiveDirective extends Directive {
+  constructor(part: Part) {
+    super();
+    if (part instanceof EventPart || part instanceof NodePart) {
+      throw new Error(
+        'The `live` directive is not allowed on text or event bindings'
```

Comment:
Why not allow this on text bindings?

Reply:
How would that work, and for what use case? We could maybe special-case strings and Nodes (but not DocumentFragment), but it'd only be useful in case someone was using contenteditable or otherwise messing with the DOM which would be pretty likely to mess up the markers. Whereas with properties, we have a relatively common use case in `<input .value>`